nautobot/docs/development/navigation-menu.md

```diff
@@ -0,0 +1,129 @@
+            ),
+        ),
+    ),
+)
+```
+
+## Modifying existing tab
+
+By defining an object with the same identifier, a developer can modify existing objects. The example below shows modifying an existing tab to have a new group.
+
+A tab object is being created with the same identifer as an existing object using the `name` attribute. Then a group is being created with a weight of `150`, which means it will appear between the already defined `Circuits` and `Provider` groups.
+
+Weights for already existing items can be found in the nautobot source code (in `navigation.py`) or with a web session open to your nautobot instance, you can inspect an element of the navbar using the developer tools. Each type of element will have an attribute `data-{type}-weight`. The type can be `tab`, `group`, `item` or `button`.
+
+This pattern works for modifying all objects in the tree. New items can be added to groups and new buttons can be added to items.
+
+``` python
+menu_tabs = (
+    NavMenuTab(
+        name="Circuits",
+        groups=(
+            NavMenuGroup(
+                name="Dummy Circuit Group",
+                weight=150,
+                items=(
+                    NavMenuItem(
+                        link="plugins:dummy_plugin:dummymodel_list",
+                        link_text="Dummy Model",
+                        permissions=[
+                            "dummy_plugin.view_dummymodel"
+                        ],
+                        buttons=(
+                            NavMenuAddButton(
+                                link="plugins:dummy_plugin:dummymodel_add",
+                                permissions=[
+                                    "dummy_plugin.add_dummymodel",
+                                ],
+                            ),
+                            NavMenuImportButton(
+                                link="plugins:dummy_plugin:dummymodel_import",
+                                permissions=[
+                                    "dummy_plugin.add_dummymodel"
+                                ],
+                            ),
+                        ),
+                    ),
+                ),
+            ),
+        ),
+    ),
+)
+```
+
+## Classes and Attributes
+
+A `NavMenuGroup` has the following attributes:
+
+* `name` - Display name to be shown in navigation menu. Identifier for this object type
+* `weight` - Defines the position the object should be displayed at (optional)
+* `groups` - List or tuple of `NavMenuGroup`
+
+A `NavMenuGroup` has the following attributes:
+
+* `name` - Display name to be shown in navigation menu. Identifier for this object type
+* `weight` - Defines the position the object should be displayed at (optional)
+* `items` - List or tuple of `NavMenuItem`
+
+A `NavMenuItem` has the following attributes:
+
+* `link` - The name of the URL path to which this menu item links. Identifier for this object type
+* `link_text` - The text presented to the user
```

Comment:
Wonder if we should change this from `link_text` to `name` to be consistent with tabs and groups?

Reply:
Not sure how consistent that will be as `link` is being used as the key in the registry. All over objects uses name as the `key`

Reply:
That's mostly an implementation detail, IMO - I think it'd make the API simpler if the developer defining the menu can use `name` at all levels instead of needing to remember `name` at some levels versus `link_text` at other levels.

Reply:
This line needs to be updated now, right?